substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -984,6 +1111,31 @@ impl<T: Config> Pallet<T> {
 			}
 		}
 
+		// update the voter snapshot status.
+		VoterSnapshotStatus::<T>::mutate(|status| {
+			match (page, status.clone()) {
+				// last page, reset status for next round.
+				(0, _) => *status = SnapshotStatus::Waiting,
+
+				(_, SnapshotStatus::Waiting) | (_, SnapshotStatus::Ongoing(_)) => {
+					let maybe_last = all_voters.last().map(|(x, _, _)| x).cloned();
+
+					if let Some(ref last) = maybe_last {
+						if maybe_last == T::VoterList::iter().last() {
+							// all voters in the voter list have been consumed.
+							*status = SnapshotStatus::Consumed;
+						} else {
+							*status = SnapshotStatus::Ongoing(last.clone());
+						}
+					} else {
+						debug_assert!(*status == SnapshotStatus::Consumed);
```

Comment:
How can this be true if this arm matched on `(_, SnapshotStatus::Waiting) | (_, SnapshotStatus::Ongoing(_))`?